parquet-column/src/main/java/org/apache/parquet/column/statistics/geometry/GeometryStatistics.java

```diff
@@ -0,0 +1,168 @@
+      LogicalTypeAnnotation.Edges edges,
+      String crs,
+      ByteBuffer metadata,
+      BoundingBox boundingBox,
+      List<Covering> coverings,
+      GeometryTypes geometryTypes) {
+    this.edges = edges;
+    this.crs = crs;
+    this.metadata = metadata;
+    this.boundingBox = supportsBoundingBox() ? boundingBox : null;
+    this.coverings = supportsCovering() ? coverings : null;
+    this.geometryTypes = geometryTypes;
+  }
+
+  public GeometryStatistics(LogicalTypeAnnotation.Edges edges, String crs, ByteBuffer metadata) {
+    this(
+        edges,
+        crs,
+        metadata,
+        new BoundingBox(),
+        Arrays.asList(new EnvelopeCovering(edges, crs)),
+        new GeometryTypes());
+  }
+
+  public BoundingBox getBoundingBox() {
+    return boundingBox;
+  }
+
+  public List<Covering> getCoverings() {
+    return coverings;
+  }
+
+  public GeometryTypes getGeometryTypes() {
+    return geometryTypes;
+  }
+
+  public void update(Binary value) {
+    if (value == null) {
+      return;
+    }
+    try {
+      Geometry geom = reader.read(value.getBytes());
+      update(geom);
+    } catch (ParseException e) {
+      abort();
+    }
+  }
+
+  private void update(Geometry geom) {
+    if (supportsBoundingBox()) {
+      boundingBox.update(geom);
+    }
+    if (supportsCovering()) {
+      coverings.stream().forEach(c -> c.update(geom));
+    }
+    geometryTypes.update(geom);
+  }
+
+  /**
+   * A bounding box is a rectangular region defined by two points, the lower left
+   * and upper right corners. It is used to represent the minimum and maximum
+   * coordinates of a geometry. Only planar geometries can have a bounding box.
+   */
+  private boolean supportsBoundingBox() {
+    // Only planar geometries can have a bounding box
+    // based on the current specification
+    return edges == LogicalTypeAnnotation.Edges.PLANAR;
+  }
+
+  /**
+   * A custom WKB-encoded polygon or multi-polygon to represent a covering of
+   * geometries. For example, it may be a bounding box, or an envelope of geometries
+   * when a bounding box cannot be built (e.g. a geometry has spherical edges, or if
+   * an edge of geographic coordinates crosses the antimeridian). In addition, it can
+   * also be used to provide vendor-agnostic coverings like S2 or H3 grids.
+   */
+  private boolean supportsCovering() {
+    // This version (POC) assumes only build coverings for planar edges
+    // In case of spherical edges, no coverings are built
+    //     return edges == LogicalTypeAnnotation.Edges.PLANAR;
+    return true;
+  }
+
+  public void merge(GeometryStatistics other) {
+    Preconditions.checkArgument(other != null, "Cannot merge with null GeometryStatistics");
+    Preconditions.checkArgument(coverings.size() == other.coverings.size(), "Coverings size must be the same");
+
+    boundingBox.merge(other.boundingBox);
+    for (int i = 0; i < coverings.size(); i++) {
+      coverings.get(i).merge(other.coverings.get(i));
```

Comment:
This is hacky. Should we organize the coverings as `Map<String, Covering>` whose key is the encoding kind?

The `merge` method will be used by `ParquetRewriter` which merges several different parquet files into a single one. Those parquet files may be written by other parquet implementations so we cannot assume these coverings are in order.

Reply:
@wgtmac Sure, that sounds a good idea. However, I am not very sure what key to use in the Map<String, Covering> variable. The spec only specifies the following and use list to save them to meta as well.
```
struct Covering {
  1: required string kind;
  2: required binary value;
}
```
and 
```
GeometryStatistics
        2: optional list<Covering> coverings;
```
And I am assuming there can be multiple coverings for the same kind, so using kind does not seems to be appropriate here.

Reply:
> I am assuming there can be multiple coverings for the same kind, so using kind does not seems to be appropriate here.

That makes sense! We can choose either approach below:

1. Change the spec to use `optional map<string, Covering> coverings` to enforce unique covering of each kind.
2. In the `merge` method, we expect each kind exists only one covering to make it simpler. If any stats does not provide that kind of covering, or has more than 1 covering of the same kind, we can simply clear the aggregated covering.

I prefer 2. WDYT?

cc @jiayuasu @paleolimbot

Reply:
Either one sounds fine to me. 2 sounds a bit more neat?

Reply:
Sure, I will go for 2 and change the PR. @wgtmac I think we should also add comments to the spec to explain this.

Reply:
Added clarification to the spec. Let me know when this PR is ready for review again. Thanks!